drainer/syncer.go

```diff
@@ -472,27 +474,35 @@ ForLoop:
 			// shouldSkip is used specially for database dsyncers like tidb/mysql/oracle
 			// although we skip some ddls, but we still need to update table info
 			// ignore means whether we should should this ddl event after binlogFilter
-			var shouldSkip, ignore bool
+			var (
+				shouldSkip, ignore bool
+				stmt               ast.StmtNode
+			)
 
-			if ignore, err = skipDDLEvent(sql, schema, table, p, s.binlogFilter); err != nil {
+			if stmt, ignore, err = skipDDLEvent(sql, schema, table, p, s.binlogFilter); err != nil {
 				break ForLoop
 			} else if ignore {
 				log.Info("skip ddl by binlog event filter", zap.String("schema", schema), zap.String("table", table),
 					zap.String("sql", sql), zap.Int64("commit ts", commitTS))
 				// A empty sql force it to evict the downstream table info.
-				if s.cfg.DestDBType == "tidb" || s.cfg.DestDBType == "mysql" {
+				if s.cfg.DestDBType == "tidb" || s.cfg.DestDBType == "mysql" || s.cfg.DestDBType == "oracle" {
 					shouldSkip = true
 				} else {
 					appendFakeBinlogIfNeeded(nil, commitTS)
 					continue
 				}
+			} else if !ignore && s.cfg.DestDBType == "oracle" {
```

Comment:
I remember that drainer already set binlog filter rule to only allow `truncate` for oracle by default, why still need to do this check? 🤔